src/components/molecules/UiDropdown/_internal/UiDropdownItem.vue

```diff
@@ -23,71 +23,70 @@ import {
   inject,
   useAttrs,
 } from 'vue';
-import type {
-  PropType,
-  ComputedRef,
-} from 'vue';
+import type { ComputedRef } from 'vue';
 import equal from 'fast-deep-equal';
 import UiIcon from '../../../atoms/UiIcon/UiIcon.vue';
+import type { IconAttrsProps } from '../../../atoms/UiIcon/UiIcon.vue';
 import UiButton from '../../../atoms/UiButton/UiButton.vue';
-import type { DropdownValue } from '../UiDropdown.vue';
-import type { Icon } from '../../../../types/icon';
+import type { ButtonAttrsProps } from '../../../atoms/UiButton/UiButton.vue';
+import type { DropdownProps } from '../UiDropdown.vue';
+import type { DefineAttrsProps } from '../../../../types';
 
-const props = defineProps({
+export interface DropdownItemProps {
   /**
    * Use this props to set the value of the dropdown item.
    */
-  value: {
-    type: [
-      String,
-      Object,
-    ] as PropType<DropdownValue>,
-    default: '',
-  },
+  value?: DropdownProps['modelValue'];
   /**
    *  Use this props to pass attrs to UiIcon.
    */
-  iconItemAttrs: {
-    type: Object,
-    default: () => ({ icon: 'present' }),
-  },
+  iconItemAttrs?: IconAttrsProps;
+}
+export type DropdownItemAttrsProps = DefineAttrsProps<DropdownItemProps, ButtonAttrsProps>;
+
+const props = withDefaults(defineProps<DropdownItemProps>(), {
+  value: '',
+  iconItemAttrs: () => ({ icon: 'present' }),
 });
-const defaultProps = computed(() => ({
+const defaultProps = computed<DropdownItemProps>(() => ({
   iconItemAttrs: {
-    icon: 'present' as Icon,
+    icon: 'present',
     ...props.iconItemAttrs,
   },
 }));
-const attrs = useAttrs();
+const attrs = useAttrs() as DropdownItemAttrsProps;
```

Comment:
As far as I know, `useAttrs` cannot be a generic function, always returns type `Record<string, unknown>`

Reply:
Ah, you're right; I thought it was our custom composable function, but it's indeed imported from the `'vue'`, sorry. In this scenario, I think we should build our custom `type-guards`, which we'll then use along with this `attrs` variable, just to make sure about the type of it. Because, again, people can pass any `attrs` they want, and we need to assure TypeScript (**with type-guard**) about types of these :) For now, I think it may stay as it is implemented right now, but I believe it should be refactored in every case to narrow the type using type-guard because it's a good pattern to use them. :)  Maybe let's create a separate issue out of that? :)

Reply:
I would also stay with the current implementation for now. We will release version 0.0.6 soon, which will drop backward compatibility for props and we will remove all `useAttrs` from components. IMO there is no need to create a new issue.  WDYT ?

Reply:
Sounds great! 👍 😄